src/System.Private.CoreLib/src/System/Threading/Timer.cs

```diff
@@ -567,7 +567,7 @@ internal bool Change(uint dueTime, uint period)
                 }
                 else
                 {
-                    if (FrameworkEventSource.IsInitialized && FrameworkEventSource.Log.IsEnabled(EventLevel.Informational, FrameworkEventSource.Keywords.ThreadTransfer))
+                    if (!EventPipeController.Initializing && FrameworkEventSource.IsInitialized && FrameworkEventSource.Log.IsEnabled(EventLevel.Informational, FrameworkEventSource.Keywords.ThreadTransfer))
```

Comment:
I would add a comment here describing why the check is useful, but otherwise LGTM

Reply:
Thanks @noahfalk, done.